src/event-log/event-log-level.ts

```diff
@@ -0,0 +1,77 @@
+import type { ULID } from 'ulid';
+import type { Event, EventLog } from './event-log.js';
+
+import { base32crockford } from '@scure/base';
+import { Encoder } from '../utils/encoder.js';
+import { monotonicFactory } from 'ulid';
+import { createLevelDatabase, LevelWrapper } from '../store/level-wrapper.js';
+
+
+type EventLogLevelConfig = {
+ /**
+   * must be a directory path (relative or absolute) where
+   *  LevelDB will store its files, or in browsers, the name of the
+   * {@link https://developer.mozilla.org/en-US/docs/Web/API/IDBDatabase IDBDatabase} to be opened.
+  */
+  location: string,
+  createLevelDatabase?: typeof createLevelDatabase,
+};
+
+export class EventLogLevel implements EventLog {
+  config: EventLogLevelConfig;
+  db: LevelWrapper<string>;
+  ulid: ULID;
+
+  constructor(config?: EventLogLevelConfig) {
+    this.config = {
+      location: 'EVENTLOG',
+      createLevelDatabase,
+      ...config,
+    };
+
+    this.db = new LevelWrapper<string>({ ...this.config, valueEncoding: 'utf8' });
+    this.ulid = monotonicFactory();
```

Comment:
Sanity: I am under the impression that the ID generated will be guaranteed monotonically increasing only if you use the same exact same instance created by monotonicFactory() by passing the same initialization/seed?

ie. I tried:
```
      for (let i = 0; i < 3; i++) {
        const ulid = monotonicFactory();
        console.log(ulid(), i);
      }
```
Output doesn't appear to be monotonically increasing.

Reply:
right, so putting `const ulid = monotonicFactory();` outside of the loop will make it so the output is monotonically increasing. e.g.

```js
const ulid = monotonicFactory();
let lastUlid = ulid();

for (let i = 0; i < 3_000_000; i++) {
  const nextUlid = ulid();
  console.log(nextUlid, i);
  console.assert(nextUlid > lastUlid);

  lastUlid = nextUlid;
}
```

Reply:
@thehenrytsai any more questions on this front?

Reply:
Yes, more question(s):

`monotonicFactory();` is instantiated every time the server is restarted, which is more analogous to my code snippet no?

I am just not clear how it can/is ensuring monotonic increments over _multiple_ instances. Granted the factory is invoked at a far quicker succession than restarting a service, so as long as the instances created no less than a second apart can have monotonic increasing behavior I guess it's fine.

Reply:
@thehenrytsai bear in mind that this is `EventLogLevel`. Multiple instances is out of the question as only a single process can communicate with level at any given time. Multiple instances using `EventLogLevel` would be _net new_ DWNs. 

Restarting a single DWN would re-instantiate the monotonic factory which uses the `Date.now()` value at time of factory creation as its seed. 

Scenarios where multiple instances are communicating with the same stores should not use leveldb period.